### ISSUE TITLE
Suicide or resign?

### ISSUE THREAD

**Issue by pheasant75** [gcp/leela-zero#424]
I have noticed that in many games, one side realize its winning, so it passes.  However, the losing side will continue to play.  While this is legit to try to create issues in the opponent's space, I noticed that it start to put stones inside it's own territory.  This act will not increase its territory so I am wondering why it keep play that.  What's the worst is that it will fill its eyes till the last one.  Sometimes the winner still passes because it is confident he is winning no matter what.  Sometimes the winner will kill that group.  So I am wondering what was the logic for the loser side to fill all it's eyes.  
Below is an example.  Black move 423 block one of the only two eyes of one group.  There should be no reason to play that.
http://zero.sjeng.org/view/df33065c27aaf5dee68cab5f97b257e1dc9b86872b7e8091a2ce4869eb6aeb62

**Comment by roy7** [comment]
Black 435 was another living group with a suicide. Perhaps LZ just hasn't fully learned the 2 eye to live concept yet.

**Comment by pheasant75** [comment]
But my point is, that play does not increase (or with intention to increase) black's territory, nor does it shrink (or has intention to shrink) white's territory, why it plays this move, not resign?

**Comment by WhiteHalmos** [comment]
@pheasant75 You said it yourself, it does not change the result of the game, so why would the bot not play it?

**Comment by killerducky** [comment]
The game was obviously one played with "-r 0" option. So Black **cannot** resign. Since it can't resign and it can't win the game, it doesn't really matter what it does.

**Comment by grolich** [comment]
@pheasant75 leelaz, like AlphaGo, doesn't count territories, nor does it care about the exact lead in territory. To LZ, a loss is a loss and a win is a win, be it by 100 points or half a point.

In addition, in order to make sure LZ doesn't resign incorrectly too often, some games are played with resignation turned off ("-r 0" option). So you see games like these sometimes.

Also, less (a lot less...) extreme examples can sometimes be observed with resign turned on, because the program has yet to learn to play or estimate the game result. Just the current state of it.

tldr: count territory difference isn't useful when trying to figure out why bots like AlphaGo or LeelaZ played a move.
